src/authentication/authentication.service.ts

```diff
@@ -44,4 +54,54 @@ export class AuthenticationService {
       token: this.generateAuthToken({ email: user.email, sub: user.id }),
     }
   }
+
+  async updatePasswordWithToken(
+    email: string,
+    token: string,
+    password: string,
+  ) {
+    let user
+    try {
+      this.jwtService.verify(token)
+      user = await this.usersService.findUserByResetToken(email, token)
+    } catch (e) {
+      if (
+        e instanceof SyntaxError || // token parse failed
+        e instanceof PrismaClientKnownRequestError // token doesn't match a user
+      ) {
+        throw new BadRequestException('Invalid token')
+      } else if (e instanceof TokenExpiredError) {
+        throw new BadRequestException('Token has expired')
+      }
+
+      throw e
+    }
+
+    try {
+      return await this.usersService.updatePassword(user.id, password, true)
+    } catch (e) {
+      console.log(e)
```

Comment:
Do we _need_ to catch here? If something `throw`s out of the `UserService` I would suspect we would toss the exception in such a way that we're not just defaulting to a `BadRequest`.. I mean, it could be that Prisma failed to update the record in the DB for some reason and therefor would/should be a `500 Internal` exception, right?